From 015f7302bf45a2ca54125e646533f540a47d4083 Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Wed, 8 Jan 2025 12:09:12 -0800
Subject: [PATCH] Allow building with elogind

This makes the runtime systemd dep requirement optional when building
with -Dsystemd=true so that colord can be built with support for
libsystemd on distros that use elogind. There should be no impact here
when building colord on distros that use systemd, both libsystemd and
systemd should exist in that case.

Signed-off-by: Achill Gilgenast <achill@achill.org>
---
 contrib/session-helper/meson.build |  2 +-
 data/meson.build                   |  2 +-
 meson.build                        | 24 +++++++++++++-----------
 src/meson.build                    |  2 +-
 4 files changed, 16 insertions(+), 14 deletions(-)

diff --git a/contrib/session-helper/meson.build b/contrib/session-helper/meson.build
index f13ddc648df4..b77cb4c72057 100644
--- a/contrib/session-helper/meson.build
+++ b/contrib/session-helper/meson.build
@@ -13,7 +13,7 @@ con2 = configuration_data()
 con2.set('servicedir', libexecdir)
 
 # replace @servicedir@
-if get_option('systemd')
+if get_option('systemd') and systemd.found()
   configure_file(
     input : 'colord-session.service.in',
     output : 'colord-session.service',
diff --git a/data/meson.build b/data/meson.build
index d253e7fcc29e..aff3c9bddd08 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -18,7 +18,7 @@ con2.set('servicedir', libexecdir)
 con2.set('daemon_user', get_option('daemon_user'))
 
 # replace @servicedir@, @daemon_user@ and @localstatedir@
-if get_option('systemd')
+if get_option('systemd') and systemd.found()
   configure_file(
     input : 'colord.service.in',
     output : 'colord.service',
diff --git a/meson.build b/meson.build
index 966a9baf7a57..adda25635e46 100644
--- a/meson.build
+++ b/meson.build
@@ -127,20 +127,22 @@ if get_option('udev_rules')
 endif
 
 if get_option('systemd')
-  systemd = dependency('systemd')
+  systemd = dependency('systemd', required : false)
   libsystemd = dependency('libsystemd')
   conf.set('HAVE_SYSTEMD', '1')
   systemd_root_prefix = get_option('systemd_root_prefix')
-  if systemd_root_prefix == ''
-    systemduserunitdir = systemd.get_variable(pkgconfig: 'systemduserunitdir')
-    systemdsystemunitdir = systemd.get_variable(pkgconfig: 'systemdsystemunitdir')
-    tmpfilesdir = systemd.get_variable(pkgconfig: 'tmpfilesdir')
-    sysusersdir = systemd.get_variable(pkgconfig: 'sysusersdir')
-  else
-    systemduserunitdir = systemd.get_variable(pkgconfig: 'systemduserunitdir', pkgconfig_define: ['rootprefix', systemd_root_prefix])
-    systemdsystemunitdir = systemd.get_variable(pkgconfig: 'systemdsystemunitdir', pkgconfig_define: ['rootprefix', systemd_root_prefix])
-    tmpfilesdir = systemd.get_variable(pkgconfig: 'tmpfilesdir', pkgconfig_define: ['rootprefix', systemd_root_prefix])
-    sysusersdir = systemd.get_variable(pkgconfig: 'sysusersdir', pkgconfig_define: ['rootprefix', systemd_root_prefix])
+  if systemd.found()
+    if systemd_root_prefix == ''
+      systemduserunitdir = systemd.get_variable(pkgconfig: 'systemduserunitdir')
+      systemdsystemunitdir = systemd.get_variable(pkgconfig: 'systemdsystemunitdir')
+      tmpfilesdir = systemd.get_variable(pkgconfig: 'tmpfilesdir')
+      sysusersdir = systemd.get_variable(pkgconfig: 'sysusersdir')
+    else
+      systemduserunitdir = systemd.get_variable(pkgconfig: 'systemduserunitdir', pkgconfig_define: ['rootprefix', systemd_root_prefix])
+      systemdsystemunitdir = systemd.get_variable(pkgconfig: 'systemdsystemunitdir', pkgconfig_define: ['rootprefix', systemd_root_prefix])
+      tmpfilesdir = systemd.get_variable(pkgconfig: 'tmpfilesdir', pkgconfig_define: ['rootprefix', systemd_root_prefix])
+      sysusersdir = systemd.get_variable(pkgconfig: 'sysusersdir', pkgconfig_define: ['rootprefix', systemd_root_prefix])
+    endif
   endif
 endif
 
diff --git a/src/meson.build b/src/meson.build
index afe3b2c0a617..3c866e60d1b2 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -26,7 +26,7 @@ resources_src = gnome.compile_resources(
 )
 
 colord_extra_deps = []
-if get_option('systemd')
+if get_option('systemd') and libsystemd.found()
   colord_extra_deps += libsystemd
 endif
 
-- 
2.51.2

